pkg/cri/server/sandbox_run_linux.go

```diff
@@ -134,6 +135,19 @@ func (c *criService) sandboxContainerSpec(id string, config *runtime.PodSandboxC
 
 	// Add sysctls
 	sysctls := config.GetLinux().GetSysctls()
+	if sysctls == nil {
+		sysctls = make(map[string]string)
+	}
+	_, ipUnprivilegedPortStart := sysctls["net.ipv4.ip_unprivileged_port_start"]
+	_, pingGroupRange := sysctls["net.ipv4.ping_group_range"]
+	if nsOptions.GetNetwork() != runtime.NamespaceMode_NODE {
+		if c.config.EnableUnprivilegedPorts && !ipUnprivilegedPortStart {
+			sysctls["net.ipv4.ip_unprivileged_port_start"] = "0"
+		}
+		if c.config.EnableUnprivilegedICMP && !pingGroupRange && !userns.RunningInUserNS() {
+			sysctls["net.ipv4.ping_group_range"] = "0 2147483647"
```

Comment:
@olljanat can u explain why use 2147483647 over 4294967295

Reply:
@jonyhy96 like first message says these are borrowed directly from Docker engine code: https://github.com/moby/moby/blob/885f3c65d54f7c85461591cda8bd9dcfb5a9723f/daemon/oci_linux.go#L775-L776

Reply:
Do u know the reason why they use that as range？

Reply:
@justincormack can you comment about this on?

Reply:
from https://systemd.io/UIDS-GIDS/

> Note that the range 2147483648…4294967294 (i.e. 2^31…2^32-2) should be handled with care. Various programs (including kernel file systems, see devpts) have trouble with UIDs outside of the signed 32bit range, i.e any UIDs equal to or above 2147483648. It is thus strongly recommended to stay away from this range in order to avoid complications. This range should be considered reserved for future, special purposes.

In general I think the "negative" (yes its unsigned in theory but...) uids and gids are basically unusable.

Reply:
also 4294967295 is defintiely unusable as it is the failure return value and placeholder for dont set on many syscalls.